lib/strings.nix

```diff
@@ -126,6 +126,18 @@ rec {
   */
   makePerlPath = makeSearchPathOutput "lib" "lib/perl5/site_perl";
 
+  /* Construct an XDG path. Suitable to use as XDG_DATA_DIRS.
+  */
+  makeXdgPath = makeSearchPathOutput "xdg" "share";
```

Comment:
This should probably be named `makeDatadirPath`. I would expect XDG path to be `etc/xdg` or `share/xdg`. Also why are you using the `xdg` output, `datadir` is usually in `out`?